src/main/rendering/raytracing/BasicRayTracer.java

```diff
@@ -24,6 +26,11 @@ public BasicRayTracer(Scene scene) {
 
 	@Override
 	public Colour trace(Ray ray) {
-		return scene.geometries.intersect(ray).isEmpty() ? scene.background : scene.ambient.colour;
+		List<Intersection> intersections = scene.geometries.intersect(ray);
+		if (intersections.isEmpty()) {
+			return scene.background;
+		} else {
+			return scene.ambient.colour.add(ray.closest(intersections).geometry.material.emission);
```

Comment:
sooooo many dots, Can we add some intermediate functions, Slightly slower, but less ugly.
For example, give geometry a getEmmission() function

Reply:
i was thinking to put that in intersection. like
```suggestion
			return scene.ambient.colour.add(ray.closest(intersections).emission());
```

Reply:
:perfect: